package/kedro_viz/integrations/kedro/run_hooks.py

```diff
@@ -0,0 +1,193 @@
+"""`kedro_viz.integrations.kedro.run_hooks` defines hooks to add additional functionalities for a kedro run."""
+
+import logging
+import traceback
+from time import perf_counter
+from typing import Any, Optional, Union
+
+from kedro.framework.hooks import hook_impl
+from kedro.pipeline.node import Node as KedroNode
+
+from kedro_viz.integrations.kedro.hooks_utils import (
+    create_dataset_event,
+    generate_timestamp,
+    hash_node,
+    write_events,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class PipelineRunStatusHook:
+    """Collect and write pipeline and dataset events during Kedro runs."""
+
+    def __init__(self):
+        # Track times, events, and context for errors
+        self._node_start: dict[str, float] = {}
+        self._events: list[dict[str, Any]] = []
+        self.datasets: dict[str, Any] = {}
+        self._current_node: Optional[Any] = None
+        self._current_dataset: Optional[str] = None
+        self._current_operation: Optional[str] = None
+        self._all_nodes: list[KedroNode] = []
+        self._started_nodes: set[str] = set()
+
+    def _write_events(self) -> None:
+        """Persist events list to the project's .viz JSON file."""
+        write_events(self._events)
+
+    def _add_event(self, event: dict[str, Any], flush: bool = False) -> None:
+        """Append one event; flush to disk when requested."""
+        self._events.append(event)
+        if flush:
+            self._write_events()
+
+    def _set_context(self, dataset: str, operation: str, node: KedroNode) -> None:
+        """Save dataset I/O and node context for error handling."""
+        self._current_dataset = dataset
+        self._current_operation = operation
+        self._current_node = node
+
+    def _clear_context(self) -> None:
+        """Clear dataset I/O context after success."""
+        self._current_dataset = None
+        self._current_operation = None
+
+    @hook_impl
+    def after_catalog_created(self, catalog: Union[Any, Any]):
+        """Grab catalog datasets for size lookups."""
+        try:
+            # prefer new KedroDataCatalog
+            from kedro.io import KedroDataCatalog
+
+            if isinstance(catalog, KedroDataCatalog):
+                self.datasets = catalog.datasets
+                return
+        except ImportError:
+            pass
+        # fallback older versions
+        self.datasets = getattr(
+            catalog, "_datasets", getattr(catalog, "_data_sets", {})
+        )
+
+    @hook_impl
+    def before_pipeline_run(self, run_params: dict, pipeline) -> None:
+        """Emit start event unless this is named pipeline."""
```

Comment:
Can you elaborate this docstring? Why should the start event not be emitted when it's a named pipeline?